pkg/boot/kexec/purgatories.go

```diff
@@ -0,0 +1,309 @@
+// Copyright 2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build ignore
+
+package main
+
+type asm struct {
+	name string
+	cc   []string
+	ld   []string
+	code string
+}
+
+var asms = []asm{
+	{
+		name: "to32bit_3000",
+		cc:   []string{"x86_64-linux-gnu-gcc", "-c", "-nostdlib", "-nostdinc", "-static"},
+		ld:   []string{"ld", "-N", "-e entry64", "-Ttext=0x3000"},
+		code: `
+# This code does its best to be PIC. As in, PIC by code, not by linker.
+# assumptions:
+# we have 8 bytes of stack to do ONE call 
+# we are in shared space -- code and data in dram
+# we are in long mode
+# in case of emergency, break glass.
+// 1: jmp 1b
+
+jmp 1f
+# save space for two arguments and stack
+.align 8
+entry: .long 0
+params: .long 0
+# This can be useful for being sure things are where we think they are,
+# and ensuring we are storing correctly.
+.long 0xdeadcafe, 0x12345678
+.long 0xcafe5555, 0xabcdefaa
+.align 128
+1:
+call 1f
+1:
+popq %rax
+andl $0xfffffff0, %eax
+movl %eax, %esp
+movl $gdt, %eax
+jmp 1f
+# recursive gdt. 
+	.balign 16
+gdt:	/* 0x00 unusable segment 
+	 * 0x08 unused
+	 * so use them as the gdt ptr
+	 */
+	.word	gdt_end - gdt - 1
+	.quad	gdt
+	.word	0, 0, 0
+			
+	/* Documented linux kernel segments */
+	/* 0x10 4GB flat code segment */
+	.word	0xFFFF, 0x0000, 0x9A00, 0x00CF
```

Comment:
https://github.com/u-root/u-root/blob/d29928f4a7dd41f32742fbd87fdb190fd5e2e885/pkg/boot/multiboot/internal/trampoline/trampoline_linux_amd64.s#L17

code segment and data segment are the same as well

Reply:
not sure I get the point here. Adding a define just moves the magic numbers to a different place ... this is basically a set of well known constants, grabbed from [c]oreboot in this case. Don't understand the comment "are the same" -- done the same way? 

no big deal, adding the #define is fine, I'm just not convinced of the value of doing so. People tend to do this both ways.